test/e2e/preemption/preemption_test.go

```diff
@@ -0,0 +1,337 @@
+		err = kClient.WaitForPodUnschedulable(sleepRespPod4, time.Duration(60)*time.Second)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		// pods in root.sandbox1 can be succeeded
+		ginkgo.By("The pods in root.sandbox1 can be succeeded")
+		for _, config := range []k8s.SleepPodConfig{sleepPod1Configs, sleepPod2Configs, sleepPod3Configs} {
+			err = kClient.WaitForPodScheduled(dev, config.Name, time.Duration(30)*time.Second)
+			gomega.Ω(err).NotTo(gomega.HaveOccurred())
+		}
+	})
+
+	ginkgo.It("Verify_no_preemption_outside_fence", func() {
+		ginkgo.By("The preemption can't go outside the fence.")
+		// update config
+		cm, err := kClient.GetConfigMap(constants.ConfigMapName, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err).NotTo(HaveOccurred())
+		cm.Data[configmanager.DefaultPolicyGroup] = fmt.Sprintf(preemptionQueueYamlFormat, sleepPodMemLimit, "fence", sleepPodMemLimit)
+		_, err = kClient.UpdateConfigMap(cm, configmanager.YuniKornTestConfig.YkNamespace)
+		Ω(err).NotTo(HaveOccurred())
+
+		// Define sleepPod
+		sleepPod1Configs := k8s.SleepPodConfig{Name: "sleepjob1", NS: dev, Mem: sleepPodMemLimit, Time: 30, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod2Configs := k8s.SleepPodConfig{Name: "sleepjob2", NS: dev, Mem: sleepPodMemLimit, Time: 30, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod3Configs := k8s.SleepPodConfig{Name: "sleepjob3", NS: dev, Mem: sleepPodMemLimit, Time: 30, Optedout: true, Labels: map[string]string{"queue": "root.sandbox1"}}
+		sleepPod4Configs := k8s.SleepPodConfig{Name: "sleepjob4", NS: dev, Mem: sleepPodMemLimit, Time: 30, Optedout: true, Labels: map[string]string{"queue": "root.sandbox2"}}
+
+		// Deploy pods in root.sandbox1
+		for _, config := range []k8s.SleepPodConfig{sleepPod1Configs, sleepPod2Configs, sleepPod3Configs} {
+			ginkgo.By("Deploy the sleep pod " + config.Name + " to the development namespace")
+			sleepObj, podErr := k8s.InitSleepPod(config)
+			Ω(podErr).NotTo(gomega.HaveOccurred())
+			sleepRespPod, podErr := kClient.CreatePod(sleepObj, dev)
+			gomega.Ω(podErr).NotTo(gomega.HaveOccurred())
+
+			// Wait for pod to move to running state
+			podErr = kClient.WaitForPodBySelectorRunning(dev,
+				fmt.Sprintf("app=%s", sleepRespPod.ObjectMeta.Labels["app"]),
+				30)
+			gomega.Ω(podErr).NotTo(gomega.HaveOccurred())
+		}
+
+		// Deploy sleepjob4 pod in root.sandbox2
+		ginkgo.By("Deploy the sleep pod " + sleepPod4Configs.Name + " to the development namespace")
+		sleepObj, podErr := k8s.InitSleepPod(sleepPod4Configs)
+		Ω(podErr).NotTo(gomega.HaveOccurred())
+		sleepRespPod4, err := kClient.CreatePod(sleepObj, dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		// sleepjob4 pod can't be scheduled before pods in root.sandbox1 are succeeded
+		ginkgo.By("The sleep pod " + sleepPod4Configs.Name + " can't be scheduled")
+		err = kClient.WaitForPodUnschedulable(sleepRespPod4, time.Duration(60)*time.Second)
```

Comment:
Nit: `60 * time.Second` should work too

Reply:
Yes, updated all `time.Duration(x)*time.Second` to `x*time.Second`.